src/Foreign/Lua/Module/System.hs

```diff
@@ -205,3 +228,47 @@ with_tmpdir parentDir tmpl callback =
       -- all args given. Second value must be converted to a string.
       tmpl' <- Lua.peek (fromAnyValue tmpl)
       Temp.withTempDirectory parentDir tmpl' (invokeWithFilePath callback')
+
+--
+-- Path manipulations
+--
+
+-- | See @System.FilePath.takeDirectory
+take_directory :: FilePath -> Lua FilePath
+take_directory fp = return $ Fp.takeDirectory fp
+
+-- | See @System.FilePath.takeFilename
+take_filename :: FilePath -> Lua FilePath
+take_filename fp = return $ Fp.takeFileName fp
+
+-- | See @System.FilePath.takeExtensions
+take_extensions :: FilePath -> Lua String
+take_extensions fp = return $ Fp.takeExtensions fp
+
+-- | See @System.FilePath.dropExtension
+drop_extensions :: FilePath -> Lua String
+drop_extensions fp = return $ Fp.dropExtensions fp
+
+-- | See @System.FilePath.hasExtension
+has_extension :: FilePath -> Lua Bool
+has_extension fp = return $ Fp.hasExtension fp
+
+-- | See @System.FilePath.splitDirectories
+split_directories :: FilePath -> Lua [FilePath]
+split_directories fp = return $ Fp.splitDirectories fp
+
+-- | See @System.FilePath.joinPath
+join_path :: [FilePath] -> Lua FilePath
+join_path fps = return $ Fp.joinPath fps
+
+-- | See @System.FilePath.isRelative
+is_relative :: FilePath -> Lua Bool
+is_relative fp = return $ Fp.isRelative fp
+
+-- | See @System.FilePath.isAbsolute
+is_absolute :: FilePath -> Lua Bool
+is_absolute fp = return $ Fp.isAbsolute fp
+
+-- | See @System.FilePath.normalise
+normalise :: FilePath -> Lua FilePath
+normalise fp = return $ Fp.normalise fp
```

Comment:
Wyh is `$` needed? Its the infix operator? which evaluates `Fp.normalise fp` first and then what does `return` do?

I understood `return Fp.normalise fp` as `return(Fp.normalise(fp))`. But its not... hm

Reply:
[Hoogle](https://hoogle.haskell.org/?hoogle=%24) has your back for questions like these. Yes, it's an infix operator which just applies the arguments. It's usually used to avoid parens. The following are equivalent:

``` haskell
normalise fp = return $ Fp.normalise fp
normalise fp = return (Fp.normalise fp)
-- "point-free style", omitting the argument name
normalise = return . Fp.normalise
```

Reply:
Thx!